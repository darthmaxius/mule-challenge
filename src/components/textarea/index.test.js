import { render, fireEvent } from '@testing-library/react';
import Textarea from '.';

describe('Button', () => {
  it('Render', () => {
    const { asFragment } = render(<Textarea className="test">Textarea</Textarea>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Click', () => {
    const handler = jest.fn();
    const textarea = render(<Textarea className="test" handlerTextarea={handler} />).getByLabelText('panel-textarea');

    fireEvent.change(textarea, { target: { value: 'a depends b' } });

    expect(handler).toBeCalled();
  });
});
