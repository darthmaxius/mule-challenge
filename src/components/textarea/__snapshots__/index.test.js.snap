// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render 1`] = `
<DocumentFragment>
  <textarea
    aria-label="panel-textarea"
    class="test"
  />
</DocumentFragment>
`;
